cluster-operator/src/main/java/io/strimzi/operator/cluster/model/JmxModel.java

```diff
@@ -0,0 +1,167 @@
+ */
+package io.strimzi.operator.cluster.model;
+
+import io.fabric8.kubernetes.api.model.ContainerPort;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.OwnerReference;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.ServicePort;
+import io.fabric8.kubernetes.api.model.networking.v1.NetworkPolicyIngressRule;
+import io.strimzi.api.kafka.model.HasJmxOptions;
+import io.strimzi.api.kafka.model.KafkaJmxAuthenticationPassword;
+import io.strimzi.api.kafka.model.template.ResourceTemplate;
+import io.strimzi.operator.common.PasswordGenerator;
+import io.strimzi.operator.common.Util;
+import io.strimzi.operator.common.model.Labels;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a mode for components with enabled JMX access
+ */
+public class JmxModel {
+    protected static final String JMX_PORT_NAME = "jmx";
+    protected static final int JMX_PORT = 9999;
+
+    protected static final String JMX_USERNAME_KEY = "jmx-username";
+    protected static final String JMX_PASSWORD_KEY = "jmx-password";
+
+    protected static final String ENV_VAR_STRIMZI_JMX_ENABLED = "STRIMZI_JMX_ENABLED";
+    protected static final String ENV_VAR_STRIMZI_JMX_USERNAME = "STRIMZI_JMX_USERNAME";
+    protected static final String ENV_VAR_STRIMZI_JMX_PASSWORD = "STRIMZI_JMX_PASSWORD";
+
+    private final boolean isEnabled;
+    private final boolean isAuthenticated;
+
+    private final String namespace;
+    private final String secretName;
+    private final Labels labels;
+    private final OwnerReference ownerReference;
+
+    private final ResourceTemplate template;
+
+    /**
+     * Constructs the JMX Model for managing JMX access to Strimzi
+     *
+     * @param namespace         Namespace of the cluster with JMX access
+     * @param secretName        Name of the JMX Secret
+     * @param labels            Labels for the JMX Secret
+     * @param ownerReference    Owner reference for the JMX Secret
+     * @param specSection       Custom resource section configuring the JMX access
+     */
+    public JmxModel(String namespace, String secretName, Labels labels, OwnerReference ownerReference, HasJmxOptions specSection) {
+        this.namespace = namespace;
+        this.secretName = secretName;
+        this.labels = labels;
+        this.ownerReference = ownerReference;
+
+        if (specSection.getJmxOptions() != null)    {
+            this.isEnabled = true;
+            this.isAuthenticated = specSection.getJmxOptions().getAuthentication() != null
+                    && specSection.getJmxOptions().getAuthentication() instanceof KafkaJmxAuthenticationPassword;
+            this.template = specSection.getTemplate() != null ? specSection.getTemplate().getJmxSecret() : null;
+        } else {
+            this.isEnabled = false;
+            this.isAuthenticated = false;
+            this.template = null;
```

Comment:
wondering if the above are just the defaults for boolean and an object so maybe we could avoid this "else" branch. Unless you put it here for clarity, which could make sense.

Reply:
Well, if we want to keep the field `final`, we cannot set it to default value first and then override it. So to remove the `else` part, we would need to remove the `final` as well. So it seems better to me this way.